yt_dlp/extractor/crunchyroll.py

```diff
@@ -849,6 +888,60 @@ class CrunchyrollBetaShowIE(CrunchyrollBaseIE):
     }]
 
     def _real_extract(self, url):
-        lang, series_id = self._match_valid_url(url).group('lang', 'id')
-        return self.url_result(f'https://www.crunchyroll.com/{lang}{series_id.lower()}',
-                               CrunchyrollShowPlaylistIE.ie_key(), series_id)
+        lang, internal_id, display_id = self._match_valid_url(url).group('lang', 'id', 'display_id')
+
+        if not self._get_cookies(url).get('etp_rt'):
+            self.to_screen(f'{display_id}: Not logged in. Switching extractors. New url: https://www.crunchyroll.com/{lang}{display_id.lower()}')
+            return self.url_result(f'https://www.crunchyroll.com/{lang}{display_id.lower()}',
+                                   CrunchyrollShowPlaylistIE.ie_key(), display_id)
+
+        api_domain, bucket, params = self._get_params(lang)
+
+        series_response = self._download_json(
+            f'{api_domain}/cms/v2{bucket}/series/{internal_id}', display_id,
+            note='Retrieving series metadata',
+            query=params)
+
+        seasons_response = self._download_json(
+            f'{api_domain}/cms/v2{bucket}/seasons?series_id={internal_id}', display_id,
+            note='Retrieving season list',
+            query=params)
+
+        req_type = self._configuration_arg('type')
+        if req_type:
+            req_type = req_type[0]
+        req_num = self._configuration_arg('number')
+        if req_num:
+            req_num = list(map(int, req_num))
+        req_str = self._configuration_arg('substr')
+        if req_str:
+            req_str = req_str[0]
+
+        def entries():
+            for season in seasons_response['items']:
+                season_id = season['id']
+                if req_type == 'dub' and not season['is_dubbed']:
+                    continue
+                if req_type == 'sub' and not season['is_subbed']:
+                    continue
+                if req_num and not season['season_number'] in req_num:
+                    continue
+                if req_str and season['title'].lower().find(req_str) == -1:
+                    continue
+                episodes_response = self._download_json(
+                    f'{api_domain}/cms/v2{bucket}/episodes?season_id={season_id}', display_id,
+                    note='Retrieving episode list for %s' % season.get('slug_title'),
+                    query=params)
+                for episode in episodes_response['items']:
+                    episode_id = episode['id']
+                    episode_display_id = episode['slug_title']
+                    yield self.url_result(
+                        f'https://beta.crunchyroll.com/{lang}watch/{episode_id}/{episode_display_id}',
+                        CrunchyrollBetaIE.ie_key(), episode_id, episode['title'])
```

Comment:
Can we return more info here? like season name, episode number etc. If so, the configuration_args become unnecessary and match_filter can be used to filter seasons without much overhead

Reply:
Yes, most of the metadata is available at this point. I wasn't aware you could short-circuit the extractor using partial metadata with match_filter that way. I should go read that code...

You'd still end up fetching the episode list for every season, but given the general overhead of dealing with a playlist that's probably not too big a deal.

Reply:
hey @tejing1 I'm still waiting on this

Reply:
Yeah, I've just been busy. I will get to it, I swear >.<

Reply:
np, just making sure u didn't forget 👍